mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -93,7 +107,12 @@ private[regression] trait GeneralizedLinearRegressionBase extends PredictorParam
         Family.fromName($(family)) -> Link.fromName($(link))), "Generalized Linear Regression " +
         s"with ${$(family)} family does not support ${$(link)} link function.")
     }
-    super.validateAndTransformSchema(schema, fitting, featuresDataType)
+    val newSchema = super.validateAndTransformSchema(schema, fitting, featuresDataType)
+    if ($(linkPredictionCol).nonEmpty) {
```

Comment:
The check ```$(linkPredictionCol).nonEmpty``` can be omitted, because we already have empty check in the next line [```SchemaUtils.appendColumn```](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala#L90). This issue exists in lots of place of the ML code base. I can do some clean up in the a separate PR.